Source/monster.cpp

```diff
@@ -4000,20 +4010,10 @@ void M_SyncStartKill(int monsterId, Point position, int pnum)
 	StartMonsterDeath(monster, pnum, false);
 }
 
-void M_UpdateLeader(int monsterId)
+void M_UpdateRelations(const Monster &monster)
 {
-	assert(static_cast<size_t>(monsterId) < MaxMonsters);
-	auto &monster = Monsters[monsterId];
-
-	for (size_t j = 0; j < ActiveMonsterCount; j++) {
-		auto &minion = Monsters[ActiveMonsters[j]];
-		if (minion.leaderRelation == LeaderRelation::Leashed && minion.leader == monsterId)
-			minion.leaderRelation = LeaderRelation::None;
-	}
-
-	if (monster.leaderRelation == LeaderRelation::Leashed) {
-		Monsters[monster.leader].packSize--;
-	}
+	ReleaseMinions(monster);
+	ShrinkLeaderPacksize(monster);

```

Comment:
Looks like it only makes sens to run one or the other since a leader can't also be a minion.

```suggestion
	if (monster.leaderRelation == LeaderRelation::Leashed)
		Monsters[monster.leader].packSize--;
	else
		ReleaseMinions(monster);
```

Reply:
While in practice there's no current situation where a unique monster has both a leader and minions there's no real reason why that couldn't happen. I've updated the PR to include a guard for the ReleaseMinions call as that's a relatively expensive function but I'd rather keep them as separate conditions.

Reply:
The game distinctly has a rule that no unique can also be a minion
```cpp
	// No unique monster would be a minion of someone else!
	assert(!monster.isUnique());
```

Reply:
Only as a debug assert though? I was hoping to keep this separate as it doesn't save much to make them exclusive conditions. Fewer changes required for a mod which implements Red Vex and Black Jade as leashed minions of Lazarus as well.

Could still do an else on hasLeashedMinions, prefer not inlining the function entirely though

Reply:
nit